app/models/paper.rb

```diff
@@ -43,4 +45,10 @@ def valid_methodologies
       'case report'
     ]
   end
+
+  def find_publication
+    if self.publication.present?
+      self.publication = Publication.where(name: self.publication.name).first_or_initialize
```

Comment:
I'm still not entirely sure why you're doing this, but I think you can do this: http://guides.rubyonrails.org/active_record_querying.html#find-or-initialize-by